fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/MultiJoin.java

```diff
@@ -74,25 +91,30 @@ public Optional<Plan> reorderJoinsAccordingToConditions() {
      */
     private Plan reorderJoinsAccordingToConditions(List<Plan> joinInputs, List<Expression> conjuncts) {
         if (joinInputs.size() == 2) {
-            Set<Slot> joinOutput = getJoinOutput(joinInputs.get(0), joinInputs.get(1));
-            Map<Boolean, List<Expression>> split = splitConjuncts(conjuncts, joinOutput);
-            List<Expression> joinConditions = split.get(true);
-            List<Expression> nonJoinConditions = split.get(false);
-
+            //Set<Slot> joinOutput = getJoinOutput(joinInputs.get(0), joinInputs.get(1));
+            //Map<Boolean, List<Expression>> split = splitConjuncts(conjuncts, joinOutput);
+            //List<Expression> joinConditions = split.get(true);
+            //List<Expression> nonJoinConditions = split.get(false);
+            Pair<List<Expression>, List<Expression>> pair = JoinUtils.extractExpressionForHashTable(
+                    joinInputs.get(0).getOutput().stream().map(SlotReference.class::cast).collect(Collectors.toList()),
```

Comment:
Why must cast to `SlotReference `?

Reply:
because extractExpressionForHashTable will judge the source table of the slot. we need it's exprId.

Reply:
`getExprId` is defined in the base class `NamedExpression`, you could get exprId from Slot.